Identity documents endpoints (#699)
MewsSystems/gitbook-connector-api: operations/identitydocuments.md

<!-- AUTOMATICALLY GENERATED, DO NOT MODIFY -->
# Identity documents

## Get all identity documents

> Returns all identity documents. This operation supports [Portfolio Access Tokens](../guidelines/multi-property.md).

Reviewer: ```suggestion
Returns all identity documents for the specified customers, with additional filtering options available. This operation supports [Portfolio Access Tokens](../guidelines/multi-property.md).
```

### Request

`[PlatformAddress]/api/connector/v1/identityDocuments/getAll`

```javascript
{
  "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
  "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
  "Client": "Sample Client 1.0.0",
  "IdentityDocumentIds": [
    "e8a72a69-c20b-4278-b699-ab0400a32ecc",
    "24a3f051-49ed-411b-9384-78187f9daae6"
  ],
  "CustomerIds": [
    "fadd5bb6-b428-45d5-94f8-fd0d89fece6d",
    "bccdafd1-3e44-439d-861f-341526b597a9"
  ],
  "ChainIds": [
    "3fa85f64-5717-4562-b3fc-2c963f66afa6",
    "5fcd1933-22f2-40b9-84da-7db04cbecec2"
  ]
}
```

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |
| `ClientToken` | string | required | Token identifying the client application. |
| `AccessToken` | string | required | Access token of the client application. |
| `Client` | string | required | Name and version of the client application. |

> | `ChainIds` | array of string | optional, max 1000 items | Unique identifiers of the Chains. If not specified, the operation returns data for all chains within scope of the Access Token. |

Reviewer: My suggestion here is just for consistency with the way the other fields are described.
```suggestion
| `ChainIds` | array of string | optional, max 1000 items | Unique identifiers of `Chain`. If not specified, the operation returns data for all chains within scope of the Access Token. |
```

### Request

`[PlatformAddress]/api/connector/v1/identityDocuments/add`

```javascript
{
  "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
  "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
  "Client": "Sample Client 1.0.0",
  "IdentityDocuments": [
    {
      "CustomerId": "e8a72a69-c20b-4278-b699-ab0400a32ecc",
      "Type": "Passport",
      "Number": "123456789",
      "ExpirationDate": "2040-11-20T00:00:00Z",
      "IssuanceDate": "2020-11-20T00:00:00Z",
      "IssuingCountryCode": "CZ",
      "IssuingCity": "Prague",
      "IdentityDocumentSupportNumber": "S-123456"
    }
  ],
  "ChainId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
}
```

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |
| `ClientToken` | string | required | Token identifying the client application. |
| `AccessToken` | string | required | Access token of the client application. |
| `Client` | string | required | Name and version of the client application. |
| `ChainId` | string | optional | Unique identifier of the chain. Required when using [Portfolio Access Tokens](../guidelines/multi-property.md), ignored otherwise. |

> | `IdentityDocuments` | array of [Identity documents parameters](identitydocuments.md#identity-documents-parameters) | required, max 10 items | Identity documents to be added |

Reviewer: ```suggestion
| `IdentityDocuments` | array of [Identity documents parameters](identitydocuments.md#identity-documents-parameters) | required, max 10 items | Identity documents to be added. |
```

> #### Identity documents parameters

Reviewer: Minor observation, but the parameters only refer to a single identity document, so `Identity document parameters` might be more appropriate.

| `Type` | [Document type](reservations.md#document-type) | required | Type of the document. |
| `Number` | string | required | Number of the document (e.g. passport number). |
| `ExpirationDate` | string | optional | Expiration date in ISO 8601 format. |
| `IssuanceDate` | string | optional | Date of issuance in ISO 8601 format. |
| `IssuingCountryCode` | string | optional | ISO 3166-1 code of the `Country`). |
| `IssuingCity` | string | optional | City where the document was issued. |
| `IdentityDocumentSupportNumber` | string | optional | Identity document support number. Only required for Spanish identity cards in Spanish hotels. |

### Response

```javascript
{
  "IdentityDocuments": [
    {
      "Id": "e8a72a69-c20b-4278-b699-ab0400a32ecc",
      "CustomerId": "fadd5bb6-b428-45d5-94f8-fd0d89fece6d",
      "Type": "Passport",
      "Number": "M1234567",
      "ExpirationDate": "2040-10-24T00:00:00Z",
      "IssuanceDate": "2024-10-24T00:00:00Z",
      "IssuingCountryCode": "CZ",
      "IssuingCity": "Prague",
      "IdentityDocumentSupportNumber": "S-123456"
    }
  ]
}
```

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |

> | `IdentityDocuments` | array of [Identity document (ver 2024-10-25)](identitydocuments.md#identity-document-ver-2024-10-25) | required, max 10 items | Updated identity documents of customers. |

Reviewer: ```suggestion
| `IdentityDocuments` | array of [Identity document (ver 2024-10-25)](identitydocuments.md#identity-document-ver-2024-10-25) | required, max 10 items | Added identity documents. |
```

Reviewer 2: This is shared between update and add, so I will change the wording here to "Modified"

Reviewer: Ah, I see. So the description needs to be made independent of the operation that uses it, e.g. "identity documents affected by the operation" or perhaps simply "identity documents", so the object can be re-used.

| `Number` | [String update value](_objects.md#string-update-value) | optional | Number of the document (e.g. passport number or `null` if the number should not be updated). |
| `ExpirationDate` | [String update value](_objects.md#string-update-value) | optional | Expiration date in ISO 8601 format (or `null` if the expiration date should not be updated). |
| `IssuanceDate` | [String update value](_objects.md#string-update-value) | optional | Date of issuance in ISO 8601 format (or `null` if the issuance date should not be updated). |
| `IssuingCountryCode` | [String update value](_objects.md#string-update-value) | optional | ISO 3166-1 code of the `Country` (or `null` if the issuing country code should not be updated). |
| `IssuingCity` | [String update value](_objects.md#string-update-value) | optional | City where the document was issued (or `null` if the issuing city should not be updated). |
| `IdentityDocumentSupportNumber` | [String update value](_objects.md#string-update-value) | optional | Identity document support number. Only required for Spanish identity cards in Spanish hotels. |

> #### Identity document type update value

Reviewer: I think I understand why this has been done like this, but these update values are getting messy.

Reviewer 2: Agreed, unfortunately we ran into some crappy limitations of how additional attributes on UpdateValue are being handled, so in addition to generator modifications, we will also have to adjust the schema generation.

### Request

`[PlatformAddress]/api/connector/v1/identityDocuments/delete`

```javascript
{
  "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
  "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
  "Client": "Sample Client 1.0.0",
  "IdentityDocumentIds": [
    "e8a72a69-c20b-4278-b699-ab0400a32ecc",
    "2e1c6096-3a28-411d-a375-150a7350b278"
  ],
  "ChainId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
}
```

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |
| `ClientToken` | string | required | Token identifying the client application. |
| `AccessToken` | string | required | Access token of the client application. |
| `Client` | string | required | Name and version of the client application. |
| `ChainId` | string | optional | Unique identifier of the chain. Required when using [Portfolio Access Tokens](../guidelines/multi-property.md), ignored otherwise. |
| `IdentityDocumentIds` | array of string | required, max 100 items | Unique identifiers of the identity documents to be deleted. |

### Response

```javascript
{}
```

## Clear identity documents

> Deletes all identity documents for provided `Customers`. This operation supports [Portfolio Access Tokens](../guidelines/multi-property.md).

Reviewer: ```suggestion
Deletes all identity documents for the specified customers. This operation supports [Portfolio Access Tokens](../guidelines/multi-property.md).
```

### Request

`[PlatformAddress]/api/connector/v1/identityDocuments/clear`

```javascript
{
  "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
  "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
  "Client": "Sample Client 1.0.0",
  "CustomerIds": [
    "fadd5bb6-b428-45d5-94f8-fd0d89fece6d",
    "bccdafd1-3e44-439d-861f-341526b597a9"
  ],
  "ChainId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
}
```

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |
| `ClientToken` | string | required | Token identifying the client application. |
| `AccessToken` | string | required | Access token of the client application. |
| `Client` | string | required | Name and version of the client application. |
| `ChainId` | string | optional | Unique identifier of the chain. Required when using [Portfolio Access Tokens](../guidelines/multi-property.md), ignored otherwise. |

> | `CustomerIds` | array of string | required, max 10 items | Unique identifiers of the customer for whom documents will be deleted. |

Reviewer: Personally, I don't mind either way of writing 'customer', but it's more important to be consistent. So I'm interpreting the new style as normal lower case in operation descriptions, but in property tables we use the notation `Customer` if we are referring to what I would call a domain entity.
```suggestion
| `CustomerIds` | array of string | required, max 10 items | Unique identifiers of the `Customer` for whom documents will be deleted. |
```
